sdk/core/core-http/review/core-http.api.md

```diff
@@ -0,0 +1,944 @@
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    body?: any;
+    clone(): WebResource;
+    decompressResponse?: boolean;
+    // (undocumented)
+    formData?: any;
+    // (undocumented)
+    headers: HttpHeadersLike;
+    // (undocumented)
+    keepAlive?: boolean;
+    // (undocumented)
+    method: HttpMethods;
+    onDownloadProgress?: (progress: TransferProgressEvent) => void;
+    onUploadProgress?: (progress: TransferProgressEvent) => void;
+    operationResponseGetter?: (operationSpec: OperationSpec, response: HttpOperationResponse) => undefined | OperationResponse;
+    // (undocumented)
+    operationSpec?: OperationSpec;
+    prepare(options: RequestPrepareOptions): WebResource;
+    // (undocumented)
+    proxySettings?: ProxySettings;
+    // (undocumented)
+    query?: {
+        [key: string]: any;
+    };
+    // (undocumented)
+    requestId: string;
+    shouldDeserialize?: boolean | ((response: HttpOperationResponse) => boolean);
+    spanOptions?: SpanOptions;
+    streamResponseBody?: boolean;
+    // (undocumented)
+    timeout: number;
+    // (undocumented)
+    url: string;
+    validateRequestProperties(): void;
+    // (undocumented)
+    withCredentials: boolean;
+}
+
+// @public (undocumented)
+export interface WebResourceLike {
+    abortSignal?: AbortSignalLike;
+    body?: any;
+    clone(): WebResourceLike;
+    decompressResponse?: boolean;
+    // (undocumented)
+    formData?: any;
+    headers: HttpHeadersLike;
+    keepAlive?: boolean;
+    method: HttpMethods;
+    onDownloadProgress?: (progress: TransferProgressEvent) => void;
+    onUploadProgress?: (progress: TransferProgressEvent) => void;
+    operationResponseGetter?: (operationSpec: OperationSpec, response: HttpOperationResponse) => undefined | OperationResponse;
+    operationSpec?: OperationSpec;
+    prepare(options: RequestPrepareOptions): WebResourceLike;
+    proxySettings?: ProxySettings;
+    query?: {
+        [key: string]: any;
+    };
+    requestId: string;
+    shouldDeserialize?: boolean | ((response: HttpOperationResponse) => boolean);
+    spanOptions?: SpanOptions;
+    streamResponseBody?: boolean;
+    timeout: number;
+    url: string;
+    validateRequestProperties(): void;
+    withCredentials: boolean;
+}
+
+
+// Warnings were encountered during analysis:
+//
+// src/policies/disableResponseDecompressionPolicy.ts:12:51 - (ae-forgotten-export) The symbol "DisableResponseDecompressionPolicy" needs to be exported by the entry point coreHttp.d.ts
+// src/policies/keepAlivePolicy.ts:24:68 - (ae-forgotten-export) The symbol "KeepAlivePolicy" needs to be exported by the entry point coreHttp.d.ts
```

Comment:
and probably this too?

Reply:
hmm, I guess the reason why the request policy types were not exported is that we would want users to use the factory methods instead of instantiate the types?

Reply:
Ah, this is the problem with having the policy functions return class references... I'm fine leaving this out for now.